keep/api/core/db_on_start.py

```diff
@@ -167,23 +169,99 @@ def try_create_single_tenant(tenant_id: str, create_default_user=True) -> None:
             logger.exception("Failed to create single tenant")
             pass
 
+def get_current_revision():
+    """Get current app revision"""
+    with engine.connect() as connection:
+        context = MigrationContext.configure(connection)
+        return context.get_current_revision()
 
-def migrate_db():
+def copy_migrations(app_migrations_path, local_migrations_path):
+    """Copy migrations to a local backup folder for safe downgrade purposes."""
+    source_versions_path = os.path.join(app_migrations_path, "versions")
+
+    # Ensure destination exists
+    os.makedirs(local_migrations_path, exist_ok=True)
```

Comment:
add try/except, in certain circumstances (e.g. openshift) this will fail to create the directories